contracts/seed/SeedFactory.sol

```diff
@@ -97,37 +99,35 @@ contract SeedFactory is CloneFactory {
       * @param _metadata              Seed contract metadata, that is IPFS URI
     */
     function deploySeed(
-        address          _admin,
+        address _admin,
         address[] memory _tokens,
         uint256[] memory _softHardThresholds,
-        uint256          _price,
-        uint256          _startTime,
-        uint256          _endTime,
-        uint32           _vestingDuration,
-        uint32           _vestingCliff,
-        bool             _isWhitelisted,
-        uint8            _fee,
-        bytes32          _metadata
-    )
-    public
-    protected
-    returns(address)
-    {
+        uint256 _price,
+        uint256 _startTime,
+        uint256 _endTime,
+        uint32 _vestingDuration,
+        uint32 _vestingCliff,
+        bool _isWhitelisted,
+        uint8 _fee,
+        bytes32 _metadata
+    ) public protected returns (address) {
         // deploy clone
         address _newSeed = createClone(address(parent));
 
         Seed(_newSeed).updateMetadata(_metadata);
 
-        {
-            // Calculating amount of Seed Token required to be transfered to deployed Seed Contract
-            uint256 requiredSeedAmount = (_softHardThresholds[1].div(_price)).mul(10**18);
-            requiredSeedAmount = requiredSeedAmount.add((requiredSeedAmount.mul(uint256(PPM))).mul(_fee).div(PPM100));
-            // checks for successful transfer of the Seed Tokens.
-            require(
-                IERC20(_tokens[0]).transferFrom(_admin, address(_newSeed), requiredSeedAmount),
-                "SeedFactory: cannot transfer seed tokens"
-            );
-        }
+        // We don't want SeedFactory to fund the Seed contract
+        // {
+        //     // Calculating amount of Seed Token required to be transfered to deployed Seed Contract
+        //     uint256 requiredSeedAmount = (_softHardThresholds[1].div(_price)).mul(10**18);
+        //     requiredSeedAmount = requiredSeedAmount
+        //.add((requiredSeedAmount.mul(uint256(PPM))).mul(_fee).div(PPM100));
+        //     // checks for successful transfer of the Seed Tokens.
+        //     require(
+        //         IERC20(_tokens[0]).transferFrom(_admin, address(_newSeed), requiredSeedAmount),
+        //         "SeedFactory: cannot transfer seed tokens"
+        //     );
+        // }
 
         // initialize
         Seed(_newSeed).initialize(
```

Comment:
it should not initialize if the contract is not funded, also here the funds are `transfer`ed from admin and not the avatar, so I would say we have to keep it

Reply:
Can you please explain this. I'm not able to understand this.

Reply:
@codebuster22 We should not be passing msg.sender as the beneficiary.  msg.sender is the Gnosis Safe wallet.  We are supposed to be taking a new param here which is the address to whom the Seed is paying fees, ie, the beneficiary.

Reply:
Okay Doug, I'll update it.